src/main/scala/amba/apb/ToTL.scala

```diff
@@ -45,23 +45,41 @@ class APBToTL()(implicit p: Parameters) extends LazyModule
       require (edgeOut.manager.minLatency >= 1)
       assert (!(in.psel && !in.penable && out.d.valid))
 
+
+      val beat = TransferSizes(1, beatBytes)
+      // The double negative here is to work around Chisel's broken implementation of widening ~x.
+      val alignedAddr =  ~(~in.paddr | (beatBytes-1).U)
+      val dataSize = UInt(log2Ceil(in.params.dataBits/8))
```

Comment:
Even though beatBytes=dataBits/8, since you use beatBytes everywhere, I think you should use it here too so it's easy to read and reason about.